packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -26,10 +27,10 @@ const setFeedId = (feedId) => { diagnostics.feedId = feedId }
 const setConnectionState = (connectionState) => { diagnostics.connection = connectionState }
 
 const setStats = (stats) => {
-  if (diagnostics.stats.length === MAX_STATS_HISTORY_SIZE) {
-    diagnostics.stats.shift()
+  if (tempStats.length === MAX_STATS_HISTORY_SIZE) {
+    tempStats.shift()
   }
-  diagnostics.stats.push(stats)
+  tempStats.push(stats)
 }
 
 const getDiagnostics = (statsCount = MAX_STATS_HISTORY_SIZE) => {
```

Comment:
We also need to check negtive statsCount like below:
```suggestion
if (statsCount > MAX_STATS_HISTORY_SIZE || statsCount <= 0)
```

Reply:
Why are you removing the main method?

Reply:
Sorry, it's not removing the signature, should be the next line, I will update it